src/api/handlers/appointments/sendReminders.js

```diff
@@ -8,75 +8,87 @@ const daysFromNow = (interval) => {
-    const template = TemplatesModel.getById(1);
-    const messageBody = TemplatesModel.generateMessage(1, language.name, {
-      patientName: appointment.patientName,
-      practitionerAddress: appointment.practitionerAddress,
-      appointmentTime: moment(appointment.appointmentTime).format("LLLL"),
-    });
+  const timeSent = new Date();
 
-    const message = {
-      appointmentId,
-      messageBody,
-      language: language.name,
-      templateName: template.templateName,
-      // UTC, will be the same for every message sent in a batch
-      timeSent: moment().format("YYYY-MM-DD HH:mm:ss"),
-    };
+  const message = {
+    appointmentId: appointment.id,
+    messageBody,
+    language: language.name,
+    templateName: template.templateName,
+    timeSent,
+  };
 
-    try {
-      await sendMessage(patientPhoneNumber, messageBody);
-    } catch (error) {
-      return {
-        error,
-        message: `Failed to send appointment reminder to ${patientPhoneNumber}`,
-      };
-    }
+  await db.transaction(async (trx) => {
+    await trx("messages").insert(message);
+    await trx("appointments")
+      .update("lastReminderSentAt", timeSent)
+      .where("appointmentId", appointment.id);
 
-    let messageId;
-    try {
-      messageId = await db("messages").insert(message);
-    } catch (error) {
-      return {
-        error,
-        message: "Could not save message to database",
-      };
-    }
-    return {
-      appointmentId,
-      messageId: messageId[0],
-    };
+    await sendMessage(appointment.patientPhoneNumber, messageBody);
   });
+};
+
+export const sendReminders = async () => {
+  // Select appointments coming up in the next 24 hours that
+  // have not had a notification in the previous 24 hours.
+
+  const start = daysFromNow(0);
+  const end = daysFromNow(1);
+  const notAfter = daysFromNow(-1);
+
+  console.info(
+    `Sending reminders for appointments from ${start} to ${end} without reminder after ${notAfter}.`
+  );
 
-  return Promise.all(appointmentsPromises)
-    .then((result) => res.status(200).send(result))
-    .catch((error) =>
-      res.status(500).send({
-        error,
-        message: `Could not send reminders for ${daysFromNow(1)}`,
-      })
-    );
+  const appointments = await db("appointments")
+    .select(
+      "appointmentId as id",
+      "language",
+      "appointmentTime",
+      "patientPhoneNumber",
+      "practitionerAddress",
+      "specialNotes",
+      "description"
+    )
+    .where("isDeleted", false)
+    .where("appointmentTime", ">=", start)
+    .where("appointmentTime", "<", end)
+    .andWhere((db) => {
+      db.where("lastReminderSentAt", "<=", notAfter).orWhereNull(
+        "lastReminderSentAt"
+      );
+    });
+
+  const promises = appointments.map(sendReminder);
+
+  await Promise.all(promises);
+};
+
+export default async (req, res) => {
+  try {
+    await sendReminders();
+    res.status(200).end();
```

Comment:
Could return here

Reply:
I have added the return although it is superfluous.